src/rest-server/src/models/token.js

```diff
@@ -67,30 +68,42 @@ const sign = async (username, application, expiration) => {
  */
 const purge = async (data, jobSpecific) => {
   const result = {};
+
+  if (jobSpecific) {
+    // get non-completed frameworks from db
+    const frameworkNamesEncoded = Object.keys(data)
+      .map(deserialize)
+      .map(encodeName);
```

Comment:
Can we avoid encodeName here? md5 encoding is cpu-intensive.

Reply:
> Can we avoid encodeName here? md5 encoding is cpu-intensive.

Fixed, thx